src/Microsoft.OData.Core/UriParser/Binders/SelectExpandBinder.cs

```diff
@@ -34,6 +33,11 @@ internal sealed class SelectExpandBinder
         /// </summary>
         private readonly IEdmNavigationSource navigationSource;
 
+        /// <summary>
+        /// The navigation source at the resource path.
+        /// </summary>
+        private readonly IEdmNavigationSource resourcePathNavigationSource;

```

Comment:
>resourcePathNavigationSource [](start = 46, length = 28)

can the root navigation source figure out from "ODataUriParserConfiguration" or other place?

Reply:
We are setting it from the `ODataPathInfo`. Having the property in `SelectExpandBinder` allows us to pass it to Bind Methods since it is not stored in the `SelectTermToken` and `ExpandTermToken`